### DISCUSSION TITLE
ur-rs: A Rust implementation of uniform resources

### DISCUSSION THREAD

**Discussion by dspicher** [BlockchainCommons/Gordian-Developer-Community#20] [Show and tell]
I'm happy to announce a Rust implementation of uniform resources:
https://github.com/dspicher/ur-rs

It still has a couple of rough edges (see the open issues), but it already passes all unit tests from the C++ and Swift implementations.

Of course, both feedback and contributions in the form of PRs are highly welcome.

**Comment by ChristopherA** [comment]
Fantastic!

What other rust-based libraries are you supporting / using with bitcoin apps? We have a rust section in our "Learning Bitcoin from the Command Line" but clearly it is an emerging language platform for Bitcoin, and makes me much happier than Python or Javascript.

-- Christopher Allen

**Comment by dspicher** [comment]
Yes, there is a lot of momentum behind Rust for Bitcoin! 

The Bitcoin Dev Kit is a very promising crate for wallet functionality: https://github.com/bitcoindevkit/bdk
Another personal favorite of mine is https://github.com/LLFourn/secp256kfun for the educational experience.

I really hope I will also find the time to implement LifeHash in Rust, it's such a cool idea!

**Comment by wolfmcnally** [comment]
Awesome! Very happy to see your Rust implementation of UR and would love to see a Rust implementation of LifeHash as well!
